### PR TITLE
New tweaks

### PR DESCRIPTION
**Pull request by amrfaissal**
- Added ANSI color writer
- Tweaked the Bee banner
- Added enhanced logging while using bee to scaffold code
- Fixed some typos and errors
- Added more error handling
- Added more docs
- Added support for multiple paths in `GOPATH`

**Screenshots**
---
bee1
bee2
bee3

### CODE REVIEW

File: apiapp.go [hunk 1]
```diff
@@ -571,21 +573,21 @@ func createapi(cmd *Command, args []string) int {
 	ColorLog("[INFO] Creating API...\n")
 
 	os.MkdirAll(apppath, 0755)
-	fmt.Println("\tcreate\t", apppath)
+	fmt.Fprintf(w, "\t%s%screate%s\t %s%s\n", "\x1b[32m", "\x1b[1m", "\x1b[21m", apppath, "\x1b[0m")
```
Comment (astaxie) [inline apiapp.go]: I don't think these is window compatible

**Comment by amrfaissal** [comment]
@astaxie The `new` command doesn't work with multiple paths in `$GOPATH`. I think we should use the `GetGOPATHs()` and use the first path by default.
bee1

**Comment by astaxie** [comment]
> I think we should use the GetGOPATHs() and use the first path by default.

agree

File: apiapp.go (hunk 1 above)
Comment (amrfaissal) [inline apiapp.go]: @astaxie It works (see my last comment). I'll submit a fix to the color writer for Windows.

Comment (amrfaissal) [inline apiapp.go]: @astaxie Fix submitted.

File: g_appcode.go [hunk 2]
```diff
@@ -17,15 +17,14 @@ package main
 import (
 	"database/sql"
 	"fmt"
+	_ "github.com/go-sql-driver/mysql"
+	_ "github.com/lib/pq"
```
Comment (astaxie) [inline g_appcode.go]: does it use the goimport to fmt? why it move to the top?

Comment (amrfaissal) [inline g_appcode.go]: Sometimes I use Emacs and it sorts the imports alphabetically automatically! You can move them down if you want.

Comment (astaxie) [inline g_appcode.go]: please move to the bottom. Actually I want to follow the goimport's style:

import{
    //system package

   // third package

  // project package
}

Comment (amrfaissal) [inline g_appcode.go]: Done!

**Pull request merged**